reactor-netty-http/src/main/java/reactor/netty/http/server/AbstractHttpServerMetricsHandler.java

```diff
@@ -138,8 +154,19 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
 				ChannelOperations<?, ?> channelOps = ChannelOperations.get(ctx.channel());
 				if (channelOps instanceof HttpServerOperations) {
 					HttpServerOperations ops = (HttpServerOperations) channelOps;
-					startWrite(ops, uriTagValue == null ? ops.path : uriTagValue.apply(ops.path),
-							methodTagValue.apply(ops.method().name()), ops.status().codeAsText().toString());
+					if (!initialized) {
+						method = methodTagValue.apply(ops.method().name());
+						path = uriTagValue == null ? ops.path : uriTagValue.apply(ops.path);
+						// Always take the remote address from the operations in order to consider proxy information
+						// Use remoteSocketAddress() in order to obtain UDS info
+						remoteSocketAddress = ops.remoteSocketAddress();
+						initialized = true;
+					}
+					if (contextView == null) {
+						contextView = ops.currentContext();
```

Comment:
if I'm correct, `ops.currentContext()` corresponds `HttpServerOperations.currentContext()` which returns the `HttpServerOperations.currentContext` attribute; but this attribute is `volatile`. 

Can we manage to avoid always accessing such volatile attribute when the AbstractHttpServerMetricsHandler is not a ContextAwareHttpServerMetricsHandler ?

For example, the following method could be added in the AbstractHttpServerMetricsHandler:

```
	protected ContextView currentContext(HttpServerOperations ops) {
		return Context.empty();
	}
```

and this method could be overriden only by the ContextAwareHttpServerMetricsHandler class:

```
	@Override
	protected final ContextView currentContext(HttpServerOperations ops) {
		return ops.currentContext();
	}
```

and instead of calling `contextView = ops.currentContext();`, the AbstractHttpServerMetricsHandler could then invoke `contextView = currentContext(ops);` ?

Reply:
Agree, please check the new commit